RetroAchievements/RAWeb
# test: add front-end coverage reporting

Nearly all of this PR's diff is _package-lock.json_ noise.

---

This PR enables front-end test coverage reporting, both in the terminal and via the Vitest official dashboard:

<img width="1449" alt="Screenshot 2024-10-02 at 6 15 53 PM" src="https://github.com/user-attachments/assets/bc48f9b4-eea4-4d9f-a2c6-255609125ce6">

**How to test this PR:**
* `sail npm install`
* `sail npm run test:coverage`

This should cause tests to begin running and a tiny Vitest service to start at http://localhost:51204/__vitest__/.

Manually navigate your browser to http://localhost:51204/__vitest__/.

If `npm run test:coverage` is run without Sail, the browser window automatically pops. Needing to do manual navigation is a Sail limitation.

After tests are done running, coverage should also appear in the terminal itself. If you navigate to the URL above, you can click a button to view coverage:

<img width="432" alt="Screenshot 2024-10-02 at 6 15 39 PM" src="https://github.com/user-attachments/assets/24db7195-d68e-4651-84ef-1e299bfb89f8">
